File: 2847-find-maximum-number-of-string-pairs/2847-find-maximum-number-of-string-pairs.cpp
class Solution {
public:
    int maximumNumberOfStringPairs(vector<string>& words)
    {

        int count = 0;
     for(int i = 0; i < words.size(); i++)
     {
         for(int j = i + 1; j < words.size(); j++)
         {
             string revstring = words[j];
            reverse(revstring.begin(),revstring.end());
             if(words[i] == revstring)
             {
                 count++;
             }
         }
     }


     return count;   
    }
};

File: 2847-find-maximum-number-of-string-pairs/2847-find-maximum-number-of-string-pairs.java
class Solution {
    public int maximumNumberOfStringPairs(String[] words) {
        // HashMap<String, Integer> map = new HashMap<String, Integer>();
        int n = words.length;
        boolean[] arr = new boolean[words.length];
        Arrays.fill(arr,false);
        int res = 0;

        for(int i = 0; i < n; i++)
        {
            if(arr[i])
            {
                continue;
            }
            for(int j = 0; j < n; j++)
            {
                if(i!=j)
                {
                    StringBuilder sb = new StringBuilder(words[j]);
                    if(words[i].equals(sb.reverse().toString()))
                    {
                        res++;
                        arr[i] = true;
                        arr[j] = true;
                    }
                }
            }
        }

        return res;
    }
}